### ISSUE TITLE
NPE when partition by EXTRACTJSONFIELD on a JSON format key field

### ISSUE THREAD

**Issue by OoRobFuoO** [confluentinc/ksql#6688]
**Describe the bug**
ksqlDb reports a NullPointerException when a stream tries to rekey a json-based key on a tombstone message.

**To Reproduce**
Version KSQL: cp-ksqldb-server 6.0

1) Create a simple stream 
`create stream s2 (MsgKey string key, f1 string, f2 string) with (kafka_topic='testTopic', value_format='json', partitions=1);`

2) Produce sample message into the Topic
```
kafkacat -P -t testTopic_2  -b localhost:9092 -K"#" -Z << EOF

{"k1":"message1","k2":"msg1"}#{"f1":"message1","f2":"msg1"}
{"k1":"message2","k2":"msg2"}#{"f1":"message2","f2":"msg2"}
{"k1":"message1","k2":"msg1"}#

```

3) Create the following CSAS:
```
SET 'auto.offset.reset'='earliest';
create stream rekeyJsonKey2 with (kafka_topic='testTopic_rekey') as select * from s2 partition by EXTRACTJSONFIELD(MsgKey, '$.k2');

```
3. Any SQL statements you ran

**Expected behavior**
Stream to process without failure as ksqldb should ignore the tombstone message.
This is a similar bug as reported in https://github.com/confluentinc/ksql/issues/6687

**Actual behaviour**
Query goes to ERROR state when running `SHOW QUERIES`
{
  "@type": "queries",
  "statementText": "show queries;",
  "queries": [
    {
      "queryString": "CREATE STREAM REKEYJSONKEY2 WITH (KAFKA_TOPIC='testTopic_rekey', PARTITIONS=1, REPLICAS=2) AS SELECT *\nFROM S2 S2\nPARTITION BY EXTRACTJSONFIELD(S2.MSGKEY, '$.k2')\nEMIT CHANGES;",
      "sinks": [
        "REKEYJSONKEY2"
      ],
      "sinkKafkaTopics": [
        "testTopic_rekey"
      ],
      "id": "CSAS_REKEYJSONKEY2_0",
      "statusCount": {
        "ERROR": 1
      },
      "queryType": "PERSISTENT",
      "state": "ERROR"
    }
  ],
  "warnings": []
}

**Additional context**
Exception Stacktrace:
2020-11-30 00:40:46,141] ERROR Unhandled exception caught in streams thread _confluent-ksql-default_query_CSAS_REKEYJSONKEY_0-b7408d84-4692-45cd-a6f1-ebaa63cc2ee8-StreamThread-4. (io.confluent.ksql.util.QueryMetadata)
org.apache.kafka.streams.errors.StreamsException: Exception caught in process. taskId=0_0, processor=KSTREAM-SOURCE-0000000000, topic=testTopic_2, partition=0, offset=2, stacktrace=java.lang.NullPointerException
	at io.confluent.ksql.execution.streams.PartitionByParamsFactory.lambda$buildMapper$2(PartitionByParamsFactory.java:166)
	at org.apache.kafka.streams.kstream.internals.KStreamMap$KStreamMapProcessor.process(KStreamMap.java:41)
	at org.apache.kafka.streams.processor.internals.ProcessorNode.lambda$process$2(ProcessorNode.java:142)
	at org.apache.kafka.streams.processor.internals.metrics.StreamsMetricsImpl.maybeMeasureLatency(StreamsMetricsImpl.java:836)
	at org.apache.kafka.streams.processor.internals.ProcessorNode.process(ProcessorNode.java:142)
	at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.forward(ProcessorContextImpl.java:236)
	at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.forward(ProcessorContextImpl.java:216)
	at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.forward(ProcessorContextImpl.java:168)
	at org.apache.kafka.streams.kstream.internals.KStreamTransformValues$KStreamTransformValuesProcessor.process(KStreamTransformValues.java:64)
	at org.apache.kafka.streams.processor.internals.ProcessorNode.lambda$process$2(ProcessorNode.java:142)
	at org.apache.kafka.streams.processor.internals.metrics.StreamsMetricsImpl.maybeMeasureLatency(StreamsMetricsImpl.java:836)
	at org.apache.kafka.streams.processor.internals.ProcessorNode.process(ProcessorNode.java:142)
	at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.forward(ProcessorContextImpl.java:236)
	at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.forward(ProcessorContextImpl.java:216)
	at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.forward(ProcessorContextImpl.java:168)
	at org.apache.kafka.streams.processor.internals.SourceNode.process(SourceNode.java:96)
	at org.apache.kafka.streams.processor.internals.StreamTask.lambda$process$1(StreamTask.java:679)
	at org.apache.kafka.streams.processor.internals.metrics.StreamsMetricsImpl.maybeMeasureLatency(StreamsMetricsImpl.java:836)
	at org.apache.kafka.streams.processor.internals.StreamTask.process(StreamTask.java:679)
	at org.apache.kafka.streams.processor.internals.TaskManager.process(TaskManager.java:1033)
	at org.apache.kafka.streams.processor.internals.StreamThread.runOnce(StreamThread.java:690)
	at org.apache.kafka.streams.processor.internals.StreamThread.runLoop(StreamThread.java:551)
	at org.apache.kafka.streams.processor.internals.StreamThread.run(StreamThread.java:510)

	at org.apache.kafka.streams.processor.internals.StreamTask.process(StreamTask.java:696)
	at org.apache.kafka.streams.processor.internals.TaskManager.process(TaskManager.java:1033)
	at org.apache.kafka.streams.processor.internals.StreamThread.runOnce(StreamThread.java:690)
	at org.apache.kafka.streams.processor.internals.StreamThread.runLoop(StreamThread.java:551)
	at org.apache.kafka.streams.processor.internals.StreamThread.run(StreamThread.java:510)
Caused by: java.lang.NullPointerException
	at io.confluent.ksql.execution.streams.PartitionByParamsFactory.lambda$buildMapper$2(PartitionByParamsFactory.java:166)
	at org.apache.kafka.streams.kstream.internals.KStreamMap$KStreamMapProcessor.process(KStreamMap.java:41)
	at org.apache.kafka.streams.processor.internals.ProcessorNode.lambda$process$2(ProcessorNode.java:142)
	at org.apache.kafka.streams.processor.internals.metrics.StreamsMetricsImpl.maybeMeasureLatency(StreamsMetricsImpl.java:836)
	at org.apache.kafka.streams.processor.internals.ProcessorNode.process(ProcessorNode.java:142)
	at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.forward(ProcessorContextImpl.java:236)
	at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.forward(ProcessorContextImpl.java:216)
	at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.forward(ProcessorContextImpl.java:168)
	at org.apache.kafka.streams.kstream.internals.KStreamTransformValues$KStreamTransformValuesProcessor.process(KStreamTransformValues.java:64)
	at org.apache.kafka.streams.processor.internals.ProcessorNode.lambda$process$2(ProcessorNode.java:142)
	at org.apache.kafka.streams.processor.internals.metrics.StreamsMetricsImpl.maybeMeasureLatency(StreamsMetricsImpl.java:836)
	at org.apache.kafka.streams.processor.internals.ProcessorNode.process(ProcessorNode.java:142)
	at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.forward(ProcessorContextImpl.java:236)
	at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.forward(ProcessorContextImpl.java:216)
	at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.forward(ProcessorContextImpl.java:168)
	at org.apache.kafka.streams.processor.internals.SourceNode.process(SourceNode.java:96)
	at org.apache.kafka.streams.processor.internals.StreamTask.lambda$process$1(StreamTask.java:679)
	at org.apache.kafka.streams.processor.internals.metrics.StreamsMetricsImpl.maybeMeasureLatency(StreamsMetricsImpl.java:836)
	at org.apache.kafka.streams.processor.internals.StreamTask.process(StreamTask.java:679)
	... 4 more

**Comment by mikebin** [comment]
Duplicate of https://github.com/confluentinc/ksql/issues/6506

**Comment by spena** [comment]
This is fixed in 0.14
https://github.com/confluentinc/ksql/pull/6508
